### DISCUSSION TITLE
[@types/geojson] - The Position type is wrong

### DISCUSSION THREAD

**Discussion by oddvarogreid** [DefinitelyTyped/DefinitelyTyped#67773] [Issues with a @types package]
### Discussed in PR #21590

According to https://datatracker.ietf.org/doc/html/rfc7946#section-3.1.1
"A position is an array of numbers. There MUST be two or more elements.  The first two elements are longitude and latitude, or easting and northing, precisely in that order and using decimal numbers.  Altitude or elevation MAY be included as an optional third element."

Based on this the type should be.

`export type Position = [number, number] | [number, number, number];`

But comments in the PR #21590 states that the change was to drastic.

It's been 5 years. It's time to fix this.

What about creating sub types for Position to avoid breaking anything?

```
export StrictPosition = [number, number] | [number, number, number];

export NonStrictPosition number[];

export Position = StrictPosition | NonStrictPosition;
```

At some point the number[] type should be removed because it's wrong.
It should not be possible to create a position with less than 2 and more than 3 coordinates.

In some cases the Position type in @types/geojson doesn't seem to work with other libraries because they use the correct type.
@mapbox/geojson-types also sets their `type GeoJSONPosition = [number, number] | [number, number, number]`.

[a comment by an automated account is not shown]

**Comment by icholy** [comment]
Nothing has changed since the discussion 5 years ago.

**Comment by oddvarogreid** [comment]
> Nothing has changed since the discussion 5 years ago.

Are you saying there is no will to change this because:
1. > 👎 👎 That's going to break every single Type definition, this change is a bit too drastic 😨
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/21590#discussion_r151582350

Even with a gentle approach using `number[] | [number, number] | [number, number, number]`

2. > * `Position` should be defined as `number[]` since no one defines their coordinates using `[number, number] | [number, number, number]`.
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/21590#issuecomment-345120672

Which is a false statement.

**Reply by icholy** [reply]
It will break code and it's not ergonomic. The "gentle" approach adds no value.

**Reply by oddvarogreid** [reply]
I get that you want to be pragmatic about this since it affects many users.
